### PR TITLE
[SG-808] Only show 'remove account' option if user is in Accounts list

### PR DESCRIPTION
**Pull request by differsthecat**
## Type of change
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
Only show 'remove account' if the account already exists in the accounts list

## Code changes

* **src/App/Pages/Accounts/LoginPageViewModel.cs:** Add `SavedEmail` property that checks if the email belongs to an account in the accounts list and only show `remove account` if they are there.

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/App/Pages/Accounts/LoginPageViewModel.cs [hunk 1]
```diff
@@ -140,6 +149,9 @@ public async Task InitAsync()
                 {
                     Email = await _stateService.GetRememberedEmailAsync();
                 }
+                var accountEmails = _stateService.AccountViews != null ? _stateService.AccountViews.Select(a => a.Email).ToList() : new List<string>();
+                SavedEmail = accountEmails.Contains(Email);
+
```
Comment (andrebispo5) [inline src/App/Pages/Accounts/LoginPageViewModel.cs:154]: Can be done as:
`SavedEmail = _stateService.AccountViews != null && _stateService.AccountViews.Any(e => e == Email);`

File: src/App/Pages/Accounts/LoginPageViewModel.cs [hunk 2]
```diff
@@ -286,6 +297,10 @@ public async Task RemoveAccountAsync()
             {
                 var confirmed = await _platformUtilsService.ShowDialogAsync(AppResources.RemoveAccountConfirmation,
                     AppResources.RemoveAccount, AppResources.Yes, AppResources.Cancel);
+
+                var accountEmails = _stateService.AccountViews != null ? _stateService.AccountViews.Select(a => a.Email).ToList() : new List<string>();
+                SavedEmail = accountEmails.Contains(Email);
```
Comment (andrebispo5) [inline src/App/Pages/Accounts/LoginPageViewModel.cs:302]: same as comment above

File: src/App/Pages/Accounts/LoginPageViewModel.cs [hunk 3]
```diff
@@ -247,9 +259,8 @@ public async Task LogInAsync(bool showLoading = true, bool checkForExistingAccou
 
         private async Task MoreAsync()
         {
-            var buttons = IsEmailEnabled
-                ? new[] { AppResources.GetPasswordHint }
-                : new[] { AppResources.GetPasswordHint, AppResources.RemoveAccount };
+            var buttons = IsEmailEnabled || !SavedEmail
+                ? new[] { AppResources.GetPasswordHint } : new[] { AppResources.GetPasswordHint, AppResources.RemoveAccount };
```
Comment (andrebispo5) [inline src/App/Pages/Accounts/LoginPageViewModel.cs:263]: Consider line break like it was before to make it easier to read.

**Review by andrebispo5 — CHANGES_REQUESTED** [review]

File: src/App/Pages/Accounts/LoginPageViewModel.cs (hunk 1 above)
Comment (differsthecat) [reply src/App/Pages/Accounts/LoginPageViewModel.cs:153]: fixed!

File: src/App/Pages/Accounts/LoginPageViewModel.cs (hunk 3 above)
Comment (differsthecat) [reply src/App/Pages/Accounts/LoginPageViewModel.cs:262]: Oops missed this

Comment (differsthecat) [reply src/App/Pages/Accounts/LoginPageViewModel.cs]: fixed!

**Review by andrebispo5 — APPROVED** [review]
Nice 👍

**Comment by differsthecat** [comment]
> @differsthecat @andrebispo5 shouldn't the password hint be removed from the More options? Given the new UI

@fedemkr Yes, we had some discussions about this. To remove it we would have to make much deeper modifications in order to make the options only visible in certain cases. In order to get the actual issue fixed we are doing this for now but will create an issue to fully address this.

**Pull request merged**